src/Nethermind/Nethermind.Taiko/TaikoSimplePayloadPreparationService.cs

```diff
@@ -0,0 +1,166 @@
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Evm.Tracing;
+using Nethermind.Int256;
+using Nethermind.Logging;
+using Nethermind.Merge.Plugin.BlockProduction;
+using Nethermind.Serialization.Rlp;
+using Nethermind.State;
+
+namespace Nethermind.Taiko;
+
+public class TaikoSimplePayloadPreparationService : IPayloadPreparationService
+{
+    private const int _emptyBlockProcessingTimeout = 2000;
+    private readonly SemaphoreSlim _worldStateLock = new(1);
+
+    private readonly ILogger _logger;
+    private readonly IBlockchainProcessor _processor;
+    private readonly IWorldState _worldState;
+    private readonly IL1OriginStore _l1OriginStore;
+    private readonly TxDecoder _txDecoder;
+
+    private readonly ConcurrentDictionary<string, IBlockProductionContext> _payloadStorage = new();
+
+    public TaikoSimplePayloadPreparationService(
+        IBlockchainProcessor processor,
+        IWorldState worldState,
+        IL1OriginStore l1OriginStore,
+        ILogManager logManager)
+    {
+        _logger = logManager.GetClassLogger();
+        _processor = processor;
+        _worldState = worldState;
+        _l1OriginStore = l1OriginStore;
+        _txDecoder = Rlp.GetStreamDecoder<Transaction>() as TxDecoder ?? throw new NullReferenceException(nameof(_txDecoder));
+    }
+
+    public virtual string StartPreparingPayload(BlockHeader parentHeader, PayloadAttributes payloadAttributes)
+    {
+        TaikoPayloadAttributes attrs = (payloadAttributes as TaikoPayloadAttributes)
+            ?? throw new InvalidOperationException("Payload attributes have incorrect type. Expected TaikoPayloadAttributes.");
+
+        string payloadId = payloadAttributes.GetPayloadId(parentHeader);
+
+        if (_payloadStorage.ContainsKey(payloadId))
+        {
+            if (_logger.IsInfo) _logger.Info($"Payload with the same parameters has already started. PayloadId: {payloadId}");
+        }
+        else
+        {
+            Block block = BuildBlock(parentHeader, attrs);
+            Hash256 parentStateRoot = parentHeader.StateRoot ?? throw new InvalidOperationException("Parent state root is null");
+            block = ProcessBlock(block, parentStateRoot);
+
+            // L1Origin **MUST NOT** be null, it's a required field in PayloadAttributes.
+            L1Origin l1Origin = attrs.L1Origin ?? throw new InvalidOperationException("L1Origin is required");
+
+            // Set the block hash before inserting the L1Origin into database.
+            l1Origin.L2BlockHash = block.Hash;
+
+            // Write L1Origin.
+            _l1OriginStore.WriteL1Origin(l1Origin.BlockID, l1Origin);
+            // Write the head L1Origin.
+            _l1OriginStore.WriteHeadL1Origin(l1Origin.BlockID);
+
+            // ignore TryAdd failure (it can only happen if payloadId is already in the dictionary)
+            _payloadStorage.TryAdd(payloadId, new NoBlockProductionContext(block, UInt256.Zero));
```

Comment:
Duplicated code in `TaikoPayloadPreparationService`

Reply:
Where comments?

Reply:
I deleted the code that was not being used (the other approach). We can get it from git history if needed in the future